acceptance-tests/src/test/scala/ch/renku/acceptancetests/BatchProjectRemovalSpec.scala

```diff
@@ -18,108 +18,74 @@
-      pattern matches last
-    }
-    val removeIds = toRemoveLinks map (elt => {
-      val projectUrlComps = elt getAttribute "href" split "/"
-      ProjectIdentifier(
-        namespace = projectUrlComps(projectUrlComps.size - 2),
-        slug = projectUrlComps last
-      )
-    })
-    removeIds foreach removeProject
-    go to ProjectsPage sleep (5 seconds)
-  }
+  private def `find and remove projects`: Unit =
+    `get user's projects from GitLab` foreach { case ProjectInfo(_, path, fullPath, created) =>
+      Given(s"project $path found")
 
-  private def removeProject(projectId: ProjectIdentifier): Unit = {
-    // Go to the project page to get the title
-    val projectPage = ProjectPage(projectId)
-    go to projectPage sleep (5 seconds)
-    projectPage.projectTitle match {
-      case Some(s) =>
-        And(s"found project $s to remove")
-        Then("remove project")
-        `remove project in GitLab`(projectId)
-      case None =>
-        And(s"could not get the title for $projectId")
-        Then("do not remove")
+      val days = JDuration.ofDays(7)
+      if (batchRemoveConfig.patterns.exists(_ matches path) && (created < Instant.now().minus(days))) {
+        And(s"the project matches the removal pattern and it's older than ${days.toDays} days")
+        Then("the project is removed")
+        `DELETE /knowledge-graph/projects/:path`(fullPath)
+      } else {
+        And(s"the project doesn't match the removal pattern or it's not older than ${days.toDays} days")
+        Then("project is left untouched")
+      }
     }
-  }
 
-  private case class BatchRemoveConfig(
-      batchRemove: Boolean = false,
-      pattern:     String = "test-(\\d{4})-(\\d{2})-(\\d{2})-(\\d{2})-(\\d{2})-(\\d{2})"
-  )
+  private case class BatchRemoveConfig(batchRemove: Boolean, patterns: List[Regex])
 
-  private lazy val batchRemoveConfig: Option[BatchRemoveConfig] = {
+  private lazy val batchRemoveConfig: BatchRemoveConfig = {
 
-    val batchRemove = Option(getProperty("batchRem")) orElse sys.env.get("RENKU_TEST_BATCH_REMOVE") match {
-      case Some(s) => Some(s.toBoolean)
-      case None    => None
-    }
-    val projectNamePattern = Option(getProperty("remPattern")) orElse sys.env.get("RENKU_TEST_REMOVE_PATTERN")
-
-    batchRemove match {
-      case Some(b) =>
-        projectNamePattern match {
-          case Some(p) => Some(BatchRemoveConfig(b, p))
-          case None    => Some(BatchRemoveConfig(b))
-        }
-      case None => None
-    }
+    val batchRemove =
+      Option(getProperty("batchRem"))
+        .orElse(sys.env.get("RENKU_TEST_BATCH_REMOVE"))
+        .flatMap(_.toBooleanOption)
+        .getOrElse(true)
+
+    val defaultPatterns = List(
+      "test-(\\d{4})-(\\d+)-(\\d+)-(\\d+)-(\\d+)-(\\d+).*",
+      "test-session-.*"
+    )
+
+    val projectNamePatterns = Option(getProperty("remPattern"))
+      .orElse(sys.env.get("RENKU_TEST_REMOVE_PATTERN"))
+      .map(List(_))
```

Comment:
When using properties or env variables, we can't specify multiple patterns, right?

Reply:
To be honest I think it's a bit dead feature. I believe it was created with an idea in mind that from time to time we run the tests manually in the mode allowing the removal of the projects. In fact, it's barely ever used and when it will be, the feature should work as before.

Reply:
Thanks Sekhar.
Yes, I thought it was you who created (and run) the test :-) As I mentioned in the PR desc, we don't delete projects created by a test that fails during execution so we thought a more systematic clean-up process was a good thing to do.